### PR TITLE
Implement tests for tooling

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**

### CODE REVIEW

File: tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java [hunk 1]
```diff
@@ -1,11 +1,70 @@
...
+    var copyPath = Path.of(tempDir.toString(), FILENAME);
+    var origFile = new File(origFilename);
+    try {
+      XZ.compress(new String[] { copyPath.toString() }, 1);
+      var compressedFilename = copyPath + ".xz";
+      assertThat(Files.exists(Path.of(compressedFilename)));
+      assertThat(Files.notExists(copyPath));
+      var compressedFile = new File(compressedFilename);
+      // TODO I don't understand why the values are inverted compared to what I expected them to be
+      assertThat(compressedFile.length()).isLessThan(origFile.length());
```
Comment (ilia-kebets-sonarsource) [inline tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java:40]: I expected to write:

```java
assertThat(origFile.length()).isLessThan(compressedFile.length());
```

But the comparison seems to be going the other way.

File: tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java [hunk 2]
```diff
@@ -1,11 +1,95 @@
...
+  }
+
   @Test
   void should_compress_files() {
-    System.out.println("yala");
+    var origSize = new File(origFile.toString()).length();
+    try {
+      XZ.compress(new String[] { origFile.toString() }, 1);
+      var compressedFilename = origFile + ".xz";
+      assertThat(Files.exists(Path.of(compressedFilename)));
```
Comment (saberduck) [inline tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java:40]: you are missing assertions, you have to call what do you want to assert about the argument (eg. `isTrue()` ) .

File: tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java [hunk 3]
```diff
@@ -37,25 +40,41 @@ public static void main(String[] args) throws Exception {
...
       try (
         var is = Files.newInputStream(file);
         var outfile = Files.newOutputStream(Path.of(file + ".xz"));
-        var outxz = new XZOutputStream(outfile, new LZMA2Options(COMPRESSION_LEVEL))
+        var outxz = new XZOutputStream(outfile, new LZMA2Options(compressionLevel))
       ) {
         is.transferTo(outxz);
       }
       try {
         Files.delete(file);
```
Comment (saberduck) [inline tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java:75]: this should be part of the same `try` block, we should delete only if we compress suceeds

File: tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java [hunk 4]
```diff
@@ -37,25 +40,41 @@ public static void main(String[] args) throws Exception {
...
-      } catch (Exception e) {
-        System.out.println("Error while deleting file: " + e.getMessage());
+      } catch (IOException e) {
```
Comment (saberduck) [inline tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java:76]: no need to catch the exception, just let it bubble up

File: tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java [hunk 5]
```diff
@@ -27,7 +30,7 @@
 public class XZ {
 
   // TODO: set this to 9 for maximum space saving (it will be slower, so we keep it at 1 for development)
```
Comment (saberduck) [inline tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java:32]: I suggest to keep default to the PROD settings and introduce cmd argument to set the level for the tests

File: tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java (hunk 2 above)
Comment (ilia-kebets-sonarsource) [reply tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java:40]: done

File: tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java (hunk 3 above)
Comment (ilia-kebets-sonarsource) [reply tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java:75]: done

File: tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java (hunk 4 above)
Comment (ilia-kebets-sonarsource) [reply tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java:76]: done

File: tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java [hunk 6]
```diff
@@ -1,11 +1,95 @@
...
+    origFile = tempDir.resolve("foo.txt");
+    Files.write(origFile, FILE_CONTENT.getBytes());
+  }
+
   @Test
   void should_compress_files() {
-    System.out.println("yala");
+    var origSize = new File(origFile.toString()).length();
+    try {
+      XZ.compress(new String[] { origFile.toString() }, 1);
```
[the remark this answers is not in the document]
Comment (ilia-kebets-sonarsource) [reply tools/fetch-node/src/test/java/org/sonarsource/javascript/XZTest.java]: done

File: tools/fetch-node/scripts/copy-to-plugin.mjs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+console.log(`Copying ${RUNTIMES_DIR} to ${targetDir}`);
+fs.copySync(RUNTIMES_DIR, targetDir);
+
+/**
+ * Reads the first CLI parameter
+ * If the path is relative, makes it absolute
+ *
+ * @returns
+ */
+function readTargetDirFromCLI() {
```
Comment (ilia-kebets-sonarsource) [inline tools/fetch-node/scripts/copy-to-plugin.mjs:57]: remove CLI param for target dir

File: tools/fetch-node/scripts/copy-to-plugin.mjs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+ * Copies tools/fetch-node/downloads/runtimes
+ * to
+ * targetDir/classes
+ */
+
+/**
+ * This script accepts a custom target directory
+ */
+const PARAM_DIR = readTargetDirFromCLI();
+const DEFAULT_TARGET_DIR = path.join(
```
Comment (ilia-kebets-sonarsource) [inline tools/fetch-node/scripts/copy-to-plugin.mjs:34]: move this to tools.mjs so we don't expose __dirname hack
rename tools.mjs to directories.mjs or something

File: tools/fetch-node/scripts/fetch-node.mjs [hunk 9]
```diff
@@ -24,62 +24,37 @@ import decompress from 'decompress';
 import decompressTargz from 'decompress-targz';
 import * as path from 'node:path';
 import * as stream from 'node:stream';
+import NODE_DISTROS from '../node-distros.mjs';
+import { DOWNLOAD_DIR, RUNTIMES_DIR } from './tools.mjs';
 
 /**
  * Fetches node.js runtimes and downloads them to
- * targetDir/classes/{distro.id}/node{.exe}
+ * downloads/{distro.id}/node{.exe}
```
Comment (ilia-kebets-sonarsource) [inline tools/fetch-node/scripts/fetch-node.mjs:32]: downloads/runtimes/{distro.id}/node{.exe}

File: package.json [hunk 10]
```diff
@@ -18,7 +18,7 @@
     "td": "node typedoc/scripts/setup-plugins.js && ./node_modules/.bin/typedoc --options typedoc/typedoc.js",
     "prepare": "husky install",
     "precommit": "pretty-quick --staged",
-    "fetch-node": "node tools/fetch-node/fetch-node.mjs && mvn validate && mvn -f tools/fetch-node/pom.xml package exec:java"
+    "fetch-node": "node tools/fetch-node/scripts/fetch-node.mjs && mvn validate && mvn -f tools/fetch-node/pom.xml package exec:java && node tools/fetch-node/scripts/copy-to-plugin.mjs"
```
Comment (ilia-kebets-sonarsource) [inline package.json:21]: remove `mvn validate`

**Pull request merged**